backend/internal/adapters/primary/handlers/internal/userservice.go

```diff
@@ -196,3 +196,29 @@ func (s *userService) DeclineFriendRequest(
 
 	return new(emptypb.Empty), nil
 }
+
+func (s *userService) EditOwnProfile(
+	c context.Context,
+	r *usersapi.EditProfileRequest,
+) (*emptypb.Empty, error) {
+	userID, ok := c.Value(contextkeys.UserID).(int64)
+	if !ok {
+		return nil, status.Error(codes.Internal, "unable to retrieve user ID from context")
+	}
+
+	if r.GetNewAvatar() != nil {
```

Comment:
nit, which you can feel free to ignore: i don't think you should split the use cases here. It should be a single use case, then in the repositories, you would perform 2 updates (in a single transaction). That way you won't have one failing and the other succeeding.